### ISSUE TITLE
Line continuation in As clause incorrectly parses return type as Variant

### ISSUE THREAD

**Issue by retailcoder** [x-vba/xdocgen#1]
Not sure this type of edge case is in scope, but this function's return type is documented in the JSON output `Variant`

```vb
Public Function Create(ByVal carMake As Long, _
ByVal carModel As String, ByVal _
carManufacturer As String) As _
Car
    Dim result As Car
    Set result = New ReadOnlyCar
    result.Make = carMake
    result.Model = carModel
    result.Manufacturer = carManufacturer
    Set Create = result
 End Function
```

Removing the (legal) line continuation between the `As` token and the type that follows, correctly documents the return type as `Car`.

**Comment by x-vba** [comment]
Thanks for the heads up! Didn't think about case, and when XDocGen can't find a type it defaults to Variant, which is why its giving Variant, as it isn't seeing a type due to the line continuation.

**Comment by retailcoder** [comment]
Yep! Line continuations have been a ...continuous (sorry!) source of parser bugs in Rubberduck - over the years we've fixed them all (I think), but yeah these little nasties are legal in so many places you wouldn't think to put them!

**Comment by retailcoder** [comment]
Semi-relatedly: if the data type is qualified e.g. `As VBA.Collection`, then the reported data type is qualified as well - if the code uses an inconsistent type-qualifying style, that easily leads to inconsistent documentation; this might very well be a bottomless can of worms (you don't want to implement type resolution off a regex parser, trust me!), but consider splitting the data type into its parts - the actual type name is always the last part. If there is one dot... Then the qualifier might be either a library or a module. If there are two dots, then the first identifier is the library, and the second is the module.

**Comment by x-vba** [comment]
The line continuation issue in the above example should be corrected now. Line continuations are definitely one of the most challenging things to work out when parsing
